app/Models/QuestionOption.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Model;
+
+use App\Models\Question;
+use App\Models\User;
+
+class QuestionOption extends Model
+{
+    use HasFactory;
+
+    public $timestamps = false;
+
+    protected $fillable = ['question_id', 'title', 'votes'];
+
+    public function question(): Question
+    {
+        return $this->belongsTo(Question::class)->first();
+    }
+
+    public function vote(User $user): void
+    {
+        $question=$this->question();
+        //if ($question->hasVoted($user)) throw new \Exception("user has already voted"); //had to take this out because of multi-option questions
```

Comment:
Maybe you should move this function to Questions model, providing a list of the chosen options.
Or worth considering moving this to the User model, as the user "votes".

Reply:
That might be true... but it came across my mind that then we should also have to validate there whether the options really belong to the question; maybe that would be impractical. But on the other hand, making a BelongsTo query every time is impractical either, so that speaks for moving it to the questions.

Reply:
But then if we have to check there, that also means making a query each time... I don't know.

Reply:
I don't see now that it cannot be fixed, but maybe you are right. (Just to be clear, I wanted to move the vote function, but do whatever makes sense for you.) Validation should happen in the controller's validator anyways, maybe with a custom rule here as well.